rllib/agents/qmix/model.py

```diff
@@ -15,7 +17,19 @@ def __init__(self, obs_space, action_space, num_outputs, model_config, name):
             self, obs_space, action_space, num_outputs, model_config, name
         )
         nn.Module.__init__(self)
-        self.obs_size = _get_size(obs_space)
+        self.preprocessing_network = None
+        if isinstance(obs_space, Tuple):
```

Comment:
this kind of logic always worries me.
why do we want to infer this implicitly from obs_space? is it because our grouping API only supports groups as Tuples?
why not have an explicit configuration for it?
otherwise, I feel like we should be able to mix things from a dict space perfectly fine as well?

Reply:
Hmm, Tuple means we don't use a preprocessor (otherwise, the Space would be a Box). The tuple has a len of num_agents. So it is more structured than for other algos, but you are right, we should make this more transparent.
...

Reply:
Oh, I see, something got messed up. Let me check this particular piece of code again in the QMIX model ...